BedrockCore.cpp

```diff
@@ -67,11 +70,22 @@ bool BedrockCore::peekCommand(BedrockCommand& command) {
 
     // If we get here, it means the command is fully completed.
     command.complete = true;
+
+    // Back out of the current transaction, it doesn't need to do anything.
+    _db.rollback();
+
+    // Done.
     return true;
 }
 
-bool BedrockCore::processCommand(BedrockCommand& command) {
+bool BedrockCore::processCommand(BedrockCommand& command, bool beginTransaction) {
     AutoTimer timer(command, BedrockCommand::PROCESS);
+
+    if (beginTransaction) {
```

Comment:
Additionally, another way to do this that doesn't require an extra parameter would be to just detect if we're already in a transaction and only start one if we're not.